Lib/test/test_uuid.py

```diff
@@ -313,63 +315,62 @@ def test_getnode(self):
         node2 = self.uuid.getnode()
         self.assertEqual(node1, node2, '%012x != %012x' % (node1, node2))
 
-    def _setup_for_pickle(self):
-        orig_uuid = sys.modules.get('uuid')
-        sys.modules['uuid'] = self.uuid
-
-        def restore_uuid_module():
-            if orig_uuid is not None:
-                sys.modules['uuid'] = orig_uuid
-            else:
-                del sys.modules['uuid']
-        self.addCleanup(restore_uuid_module)
-
     def test_pickle_roundtrip(self):
-        self._setup_for_pickle()
+        def pickle_roundtrip(obj, protocol):
+            return pickle.loads(pickle.dumps(obj, protocol=protocol))
+        copiers = [
+            ('copy.copy', copy.copy),
+            ('copy.deepcopy', copy.deepcopy),
+        ] + [
+            (f'pickle round-trip with protocol={protocol}',
+             partial(pickle_roundtrip, protocol=protocol))
+            for protocol in range(pickle.HIGHEST_PROTOCOL + 1)
+        ]
 
-        u = self.uuid.UUID('12345678123456781234567812345678')
-        self.assertEqual(u, pickle.loads(pickle.dumps(u)))
+        with support.swap_item(sys.modules, 'uuid', self.uuid):
+            for copy_func_name, copy_func in copiers:
+                with self.subTest(copy_func_name=copy_func_name):
+                    for is_safe in self.uuid.SafeUUID:
+                        u = self.uuid.UUID('d82579ce6642a0de7ddf490a7aec7aa5',
+                                           is_safe=is_safe)
+                        self.assertEqual(u, copy_func(u))
```

Comment:
Usually in `assertEqual()` the first argument is an actual value, and the second argument is an expected value.